src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -0,0 +1,375 @@
+-- Module: Chainweb.Pact.InMemoryCheckpointer
+-- Copyright: Copyright © 2018 Kadena LLC.
+-- License: MIT
+-- Maintainer: Emmanuel Denloye-Ito <emmanuel@kadena.io>
+-- Stability: experimental
+--
+
+module Chainweb.Pact.Backend.ChainwebPactDb
+  ( chainwebpactdb
+  , SQLiteEnv(..)
+  , sConn
+  , sConfig
+  , sLogger
+  , ReorgVersion(..)
+  , BlockVersion(..)
+  , bvVersion
+  , bvBlock
+  , detectVersionChange
+  , onVersionChange
+  , OnVersionChange(..)
+  ) where
+
+-- import Control.Concurrent.MVar
+-- import Control.Monad.State.Strict
+import Control.Lens
+import Control.Monad.Reader
+import Control.Exception
+
+-- import Data.Text.Encoding
+-- import qualified Data.Text as T
+import qualified Data.ByteString as BS
+import Data.Int
+import Data.Serialize (encode)
+import Data.String
+import Data.String.Conv
+import Data.Time.Clock
+import Data.Time.Format
+import Database.SQLite3.Direct as SQ3
+import qualified Data.Aeson as A
+
+import Prelude hiding (log)
+
+import System.IO.Extra
+
+
+-- import Test.Tasty.HUnit
+
+-- pact
+
+-- import Pact.Interpreter
+import Pact.Persist hiding (beginTx, commitTx, rollbackTx, createTable)
+import Pact.Persist.SQLite
+-- import Pact.PersistPactDb hiding (log)
+import Pact.Types.Logger hiding (log)
+import Pact.Types.Persistence
+import Pact.Types.SQLite
+import Pact.Types.Term(ModuleName(..), TableName(..))
+
+-- chainweb
+
+import Chainweb.BlockHeader
+import Chainweb.BlockHash
+
+chainwebpactdb :: PactDb p
+chainwebpactdb = PactDb
+  {
+    _readRow = readRow
+  , _writeRow = writeRow
+  , _keys = keys
+  , _txids = txids
+  , _createUserTable = createUserTable
+  , _getUserTableInfo = getUserTableInfo
+  , _beginTx = beginTx
+  , _commitTx = commitTx
+  , _rollbackTx = rollbackTx
+  , _getTxLog = getTxLog
+  }
+
+readRow :: Domain k v -> k -> Method e (Maybe v)
+readRow = undefined
+
+writeRow :: WriteType -> Domain k v -> k -> v -> Method e ()
+writeRow = undefined
+
+keys :: TableName -> Method e [RowKey]
+keys = undefined
+
+txids :: TableName -> TxId -> Method e [TxId]
+txids = undefined
+
+createUserTable :: TableName -> ModuleName -> Method e ()
+createUserTable = undefined
+
+getUserTableInfo :: TableName -> Method e ModuleName
+getUserTableInfo = undefined
+
+beginTx :: Maybe TxId -> Method e ()
+beginTx = undefined
+
+commitTx :: Method e [TxLog A.Value]
+commitTx = undefined
+
+rollbackTx :: Method e ()
+rollbackTx = undefined
+
+getTxLog :: Domain k v -> TxId -> Method e [TxLog v]
+getTxLog = undefined
+
+_initFunction :: SQLiteEnv -> IO ()
+_initFunction = undefined
+
+data SQLiteEnv = SQLiteEnv {
+    _sConn :: Database
+  , _sConfig :: SQLiteConfig
+  , _sLogger :: Logger
+  }
+
+makeLenses ''SQLiteEnv
+
+newtype ReorgVersion = ReorgVersion
+  { _getReOrgVersion :: Int64
+  }
+  deriving newtype Num
+
+data BlockVersion = BlockVersion
+  { _bvBlock :: !BlockHeight
+  , _bvVersion :: !ReorgVersion
+  }
+
+makeLenses ''BlockVersion
+
+_checkTableExist :: Database -> Utf8 -> IO [[SType]]
```

Comment:
This is useless now as is the test below, we should start moving to more meaningful tests soon.

Reply:
Deleted.